### PR TITLE
UF-5018: Created fetch-logic.

### PR DESCRIPTION
**Pull request by foobar-1337**

### CODE REVIEW

File: src/main/java/se/sundsvall/contactsettings/service/ContactSettingsService.java [hunk 1]
```diff
@@ -55,6 +62,48 @@ public ContactSetting readContactSetting(final String id) {
...
+	}
+
+	public List<ContactSetting> findByPartyIdAndFilter(final String partyId, final Map<String, List<String>> filter) {
+		final var parent = contactSettingRepository.findByPartyId(partyId)
+			.orElseThrow(() -> Problem.valueOf(NOT_FOUND, String.format(ENTITY_BY_PARTY_ID_NOT_FOUND, partyId)));
+
+		// Call the actual search-and-collect logic.
+		return searchAndCollectFromDelegateChain(parent, toListFilter(filter), new HashSet<>()).stream()
+			.map(ContactSettingMapper::toContactSetting)
+			.distinct()
```
Comment (ronnberg) [inline src/main/java/se/sundsvall/contactsettings/service/ContactSettingsService.java:85]: Is distinct needed?

Comment (foobar-1337) [reply src/main/java/se/sundsvall/contactsettings/service/ContactSettingsService.java:85]: No, not any more.

File: src/main/java/se/sundsvall/contactsettings/integration/db/ContactSettingRepository.java [hunk 2]
```diff
@@ -14,9 +15,9 @@ public interface ContactSettingRepository extends JpaRepository<ContactSettingEn
 	 * Find by partyId.
 	 *
 	 * @param partyId of the ContactSetting owner.
-	 * @return a List of ContactSettingEntity objects.
+	 * @return an Optional of ContactSettingEntity objects.
```
Comment (MickeWallner) [inline src/main/java/se/sundsvall/contactsettings/integration/db/ContactSettingRepository.java:18]: objects -> object?

**Review by MickeWallner — DISMISSED** [review]

File: src/main/java/se/sundsvall/contactsettings/integration/db/ContactSettingRepository.java (hunk 2 above)
Comment (foobar-1337) [reply src/main/java/se/sundsvall/contactsettings/integration/db/ContactSettingRepository.java:18]: yes

**Review by MickeWallner — APPROVED** [review]

**Pull request merged**